{phys}[foss/2019a] TRIQS v2.2.0 w/ Python 2.7.15

rokzitko opened the pull request:
Adding support for a new package: TRIQS, Toolbox for Research on Interacting Quantum Systems 
https://triqs.github.io/triqs/2.1.x/

boegel commented on easybuild/easyconfigs/t/TRIQS/TRIQS-2.1.1-foss-2018b.eb:
```diff
@@ -0,0 +1,44 @@
+toolchainopts = {'pic': True, 'usempi': True}
+
+sources = ['%(version)s.tar.gz']
+source_urls = ['https://github.com/TRIQS/triqs/archive/']
```
@rokzitko Please add a `sha256` checksum; you can let the `eb` command take care of that for you using `eb TRIQS-2.1.1-foss-2018b.eb --inject-checksums`

boegel commented on easybuild/easyconfigs/t/TRIQS/TRIQS-2.1.1-foss-2018b.eb:
```diff
@@ -0,0 +1,44 @@
+  ('GMP', '6.1.2'),
+  ('Mako', '1.0.7', '-Python-2.7.15'),
+  ('h5py', '2.8.0', '-Python-2.7.15'),
+  ('CMake', '3.12.1')
```
`CMake` is most likely only required to build `TRIQS`, it's not required to use it?

If so, make `CMake` a build-only dependency:

```python
builddependencies = [
    ('CMake', '3.12.1'),
]
```

boegel commented on easybuild/easyconfigs/t/TRIQS/TRIQS-2.1.1-foss-2018b.eb:
```diff
@@ -0,0 +1,44 @@
+
+separate_build_dir = True
+
+runtest = 'test'
```
Style nitpicking: `separate_build_dir = True` and `runtest = 'test'` usually go *above* `sanity_check_paths`

boegel commented on easybuild/easyconfigs/t/TRIQS/TRIQS-2.1.1-foss-2018b.eb:
```diff
@@ -0,0 +1,44 @@
+
+runtest = 'test'
+
+modextrapaths = {'PYTHONPATH': 'lib/python2.7/site-packages', 'CMAKE_PREFIX_PATH': ['lib/cmake/triqs', 'lib/cmake/cpp2py']}
```
This line is too long, and we like to have each key on a separate line anyway, so:

```python
modextrapaths = {
    'PYTHONPATH': 'lib/python%(pyshortver)s/site-packages',
    'CMAKE_PREFIX_PATH': ['lib/cmake/triqs', 'lib/cmake/cpp2py'],
}

boegel commented on easybuild/easyconfigs/t/TRIQS/TRIQS-2.1.1-foss-2018b.eb:
```diff
@@ -0,0 +1,44 @@
+source_urls = ['https://github.com/TRIQS/triqs/archive/']
+
+dependencies = [
+  ('Python', '2.7.15'),
```
Style check fails here, probably you're not indenting with 4 spaces (but just two), so please fix?

boegel commented on easybuild/easyconfigs/t/TRIQS/TRIQS-2.1.1-foss-2018b.eb:
```diff
@@ -0,0 +1,44 @@
+
+dependencies = [
+  ('Python', '2.7.15'),
+  ('Boost', '1.68.0', '-Python-2.7.15'),
```
If `Python` is included as a dependency, please indicate so by using the following `versionsuffix` (right below where you define `version`), so:

```python
version = '2.1.1'
versionsuffix = '-Python-%(pyver)s'
```

The `%(pyver)s` part is a template value, which gets automatically replaced with the Python version included as a dependency.

Any dependencies that have the same versionsuffix can then leverage `versionsuffix`, so:

```python
dependencies = [
    ('Python', '2.7.15'),
    ('Boost', '1.68.0', versionsuffix),
    ...
```

In general, you should avoid hardcoding things like the `Python` version in multiple places: just specify the version once in `dependencies` (like you did above), and then use the template values like `%(pyver)s` and `%(pyshortver)s`, see https://easybuild.readthedocs.io/en/latest/version-specific/easyconfig_templates.html#avail-easyconfig-templates

boegel commented on easybuild/easyconfigs/t/TRIQS/TRIQS-2.1.1-foss-2018b.eb:
```diff
@@ -0,0 +1,44 @@
+]
+
+sanity_check_paths = {
+    'files': ['lib/libtriqs.so'],
```
Please avoid hardcoding `.so` (since on macOS it's actually `.dylib`), use the `SHLIB_EXT` constant instead:

```python
sanity_check_paths = {
    'files': ['lib/libtriqs.%s' % SHLIB_EXT],

rokzitko commented:
Thank you for the comments and explanations, much appreciated! I’ve updated the easyconfig accordingly.

Kind regards,
Rok


> On 26 Aug 2019, at 15:56, Kenneth Hoste <notifications@github.com> wrote:
> 
> @boegel requested changes on this pull request.
> 
> Thanks a lot for your contribution @rokzitko <https://github.com/rokzitko>!
> 
> There are a couple of minor improvements that should be made to make the tests happy, and to ensure your easyconfig file can be maintained/updated easily in the future.
> 
> Please see my comments & suggestions, do let us know if you have any questions!
> 
> In easybuild/easyconfigs/t/TRIQS/TRIQS-2.1.1-foss-2018b.eb <https://github.com/easybuilders/easybuild-easyconfigs/pull/8835#discussion_r317611412>:
> 
> > +
> +homepage = 'https://triqs.github.io/'
> +description = """
> + TRIQS (Toolbox for Research on Interacting Quantum Systems) is a
> + scientific project providing a set of C++ and Python libraries to
> + develop new tools for the study of interacting quantum systems.
> +"""
> +
> +docurls = ['https://triqs.github.io/triqs/2.1.x/reference.html']
> +software_license = 'LicenseGPLv3'
> +
> +toolchain = {'name': 'foss', 'version': '2018b'}
> +toolchainopts = {'pic': True, 'usempi': True}
> +
> +sources = ['%(version)s.tar.gz']
> +source_urls = ['https://github.com/TRIQS/triqs/archive/']
> @rokzitko <https://github.com/rokzitko> Please add a sha256 checksum; you can let the eb command take care of that for you using eb TRIQS-2.1.1-foss-2018b.eb --inject-checksums
> 
> In easybuild/easyconfigs/t/TRIQS/TRIQS-2.1.1-foss-2018b.eb <https://github.com/easybuilders/easybuild-easyconfigs/pull/8835#discussion_r317611858>:
> 
> > +software_license = 'LicenseGPLv3'
> +
> +toolchain = {'name': 'foss', 'version': '2018b'}
> +toolchainopts = {'pic': True, 'usempi': True}
> +
> +sources = ['%(version)s.tar.gz']
> +source_urls = ['https://github.com/TRIQS/triqs/archive/']
> +
> +dependencies = [
> +  ('Python', '2.7.15'),
> +  ('Boost', '1.68.0', '-Python-2.7.15'),
> +  ('Clang', '7.0.1'),
> +  ('GMP', '6.1.2'),
> +  ('Mako', '1.0.7', '-Python-2.7.15'),
> +  ('h5py', '2.8.0', '-Python-2.7.15'),
> +  ('CMake', '3.12.1')
> CMake is most likely only required to build TRIQS, it's not required to use it?
> 
> If so, make CMake a build-only dependency:
> 
> builddependencies = [
>     ('CMake', '3.12.1'),
> ]
> In easybuild/easyconfigs/t/TRIQS/TRIQS-2.1.1-foss-2018b.eb <https://github.com/easybuilders/easybuild-easyconfigs/pull/8835#discussion_r317612040>:
> 
> > +  ('Boost', '1.68.0', '-Python-2.7.15'),
> +  ('Clang', '7.0.1'),
> +  ('GMP', '6.1.2'),
> +  ('Mako', '1.0.7', '-Python-2.7.15'),
> +  ('h5py', '2.8.0', '-Python-2.7.15'),
> +  ('CMake', '3.12.1')
> +]
> +
> +sanity_check_paths = {
> +    'files': ['lib/libtriqs.so'],
> +    'dirs': ['bin', 'include', 'lib', 'share'],
> +}
> +
> +separate_build_dir = True
> +
> +runtest = 'test'
> Style nitpicking: separate_build_dir = True and runtest = 'test' usually go above sanity_check_paths
> 
> In easybuild/easyconfigs/t/TRIQS/TRIQS-2.1.1-foss-2018b.eb <https://github.com/easybuilders/easybuild-easyconfigs/pull/8835#discussion_r317612543>:
> 
> > +  ('GMP', '6.1.2'),
> +  ('Mako', '1.0.7', '-Python-2.7.15'),
> +  ('h5py', '2.8.0', '-Python-2.7.15'),
> +  ('CMake', '3.12.1')
> +]
> +
> +sanity_check_paths = {
> +    'files': ['lib/libtriqs.so'],
> +    'dirs': ['bin', 'include', 'lib', 'share'],
> +}
> +
> +separate_build_dir = True
> +
> +runtest = 'test'
> +
> +modextrapaths = {'PYTHONPATH': 'lib/python2.7/site-packages', 'CMAKE_PREFIX_PATH': ['lib/cmake/triqs', 'lib/cmake/cpp2py']}
> This line is too long, and we like to have each key on a separate line anyway, so:
> 
> modextrapaths = {
>     'PYTHONPATH': 'lib/python%(pyshortver)s/site-packages',
>     'CMAKE_PREFIX_PATH': ['lib/cmake/triqs', 'lib/cmake/cpp2py'],
> }
> In easybuild/easyconfigs/t/TRIQS/TRIQS-2.1.1-foss-2018b.eb <https://github.com/easybuilders/easybuild-easyconfigs/pull/8835#discussion_r317612813>:
> 
> > + TRIQS (Toolbox for Research on Interacting Quantum Systems) is a
> + scientific project providing a set of C++ and Python libraries to
> + develop new tools for the study of interacting quantum systems.
> +"""
> +
> +docurls = ['https://triqs.github.io/triqs/2.1.x/reference.html']
> +software_license = 'LicenseGPLv3'
> +
> +toolchain = {'name': 'foss', 'version': '2018b'}
> +toolchainopts = {'pic': True, 'usempi': True}
> +
> +sources = ['%(version)s.tar.gz']
> +source_urls = ['https://github.com/TRIQS/triqs/archive/']
> +
> +dependencies = [
> +  ('Python', '2.7.15'),
> Style check fails here, probably you're not indenting with 4 spaces (but just two), so please fix?
> 
> In easybuild/easyconfigs/t/TRIQS/TRIQS-2.1.1-foss-2018b.eb <https://github.com/easybuilders/easybuild-easyconfigs/pull/8835#discussion_r317614295>:
> 
> > + scientific project providing a set of C++ and Python libraries to
> + develop new tools for the study of interacting quantum systems.
> +"""
> +
> +docurls = ['https://triqs.github.io/triqs/2.1.x/reference.html']
> +software_license = 'LicenseGPLv3'
> +
> +toolchain = {'name': 'foss', 'version': '2018b'}
> +toolchainopts = {'pic': True, 'usempi': True}
> +
> +sources = ['%(version)s.tar.gz']
> +source_urls = ['https://github.com/TRIQS/triqs/archive/']
> +
> +dependencies = [
> +  ('Python', '2.7.15'),
> +  ('Boost', '1.68.0', '-Python-2.7.15'),
> If Python is included as a dependency, please indicate so by using the following versionsuffix (right below where you define version), so:
> 
> version = '2.1.1'
> versionsuffix = '-Python-%(pyver)s'
> The %(pyver)s part is a template value, which gets automatically replaced with the Python version included as a dependency.
> 
> Any dependencies that have the same versionsuffix can then leverage versionsuffix, so:
> 
> dependencies = [
>     ('Python', '2.7.15'),
>     ('Boost', '1.68.0', versionsuffix),
>     ...
> In general, you should avoid hardcoding things like the Python version in multiple places: just specify the version once in dependencies (like you did above), and then use the template values like %(pyver)s and %(pyshortver)s, see https://easybuild.readthedocs.io/en/latest/version-specific/easyconfig_templates.html#avail-easyconfig-templates <https://easybuild.readthedocs.io/en/latest/version-specific/easyconfig_templates.html#avail-easyconfig-templates>
> In easybuild/easyconfigs/t/TRIQS/TRIQS-2.1.1-foss-2018b.eb <https://github.com/easybuilders/easybuild-easyconfigs/pull/8835#discussion_r317614602>:
> 
> > +
> +sources = ['%(version)s.tar.gz']
> +source_urls = ['https://github.com/TRIQS/triqs/archive/']
> +
> +dependencies = [
> +  ('Python', '2.7.15'),
> +  ('Boost', '1.68.0', '-Python-2.7.15'),
> +  ('Clang', '7.0.1'),
> +  ('GMP', '6.1.2'),
> +  ('Mako', '1.0.7', '-Python-2.7.15'),
> +  ('h5py', '2.8.0', '-Python-2.7.15'),
> +  ('CMake', '3.12.1')
> +]
> +
> +sanity_check_paths = {
> +    'files': ['lib/libtriqs.so'],
> Please avoid hardcoding .so (since on macOS it's actually .dylib), use the SHLIB_EXT constant instead:
> 
> sanity_check_paths = {
>     'files': ['lib/libtriqs.%s' % SHLIB_EXT],
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/easybuilders/easybuild-easyconfigs/pull/8835?email_source=notifications&email_token=AG57NBD72BK6FLNA4RYQGXLQGPOJFA5CNFSM4IPOUZWKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCCVGBEY#pullrequestreview-279601299>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AG57NBEJRASUHOME2CCNJGTQGPOJFANCNFSM4IPOUZWA>.
>

rokzitko commented:
I think all the issues have been address and that the easyconfig for TRIQS is ready to be merged.

boegel commented on easybuild/easyconfigs/t/TRIQS/TRIQS-2.1.1-foss-2018b-Python-2.7.15.eb:
```diff
@@ -0,0 +1,52 @@
+sanity_check_paths = {
+    'files': ['lib/libtriqs.%s' % SHLIB_EXT],
+    'dirs': ['bin', 'include', 'lib', 'share'],
+}
```
add a sanity check command to ensure that the Python component is installed correctly?

```python
sanity_check_commands = ["python -c 'import pytriqs'"]

boegel commented on easybuild/easyconfigs/t/TRIQS/TRIQS-2.1.1-foss-2018b-Python-2.7.15.eb:
```diff
@@ -0,0 +1,52 @@
+
+separate_build_dir = True
+
+runtest = 'test'
```
Two tests are failing for me on Intel Skylake (see upcoming test report), any idea how robust the TRIQS test suite is?

Wentzell replied:
This is fixed for the updated build script which is now based on foss/2019a

boegel commented:
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
node3100.skitty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/32624f9f642eca1e558cf28cacfdac15 for a full test report.

boegel commented:
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
generoso - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2660 v4 @ 2.00GHz, Python 3.6.8
See https://gist.github.com/0081ea20ea9b54b5238807bc24d45aa8 for a full test report.

boegel commented:
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2628.swalot.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/6317180842017d78edb3173c0e89c54d for a full test report.

boegel commented:
@rokzitko Latest test reports show that the tests work just fine on Intel Haswell and Broadwell, so the problem seems specific to Intel Skylake.
Any idea what the cause could be?

If this is a known problem with TRIQS and there's an open issue to track it, I guess this doesn't need to block this PR...

rokzitko commented:
I’ve tested on our Sandy Bridge, as well as on Opteron 63xx and Epyc 1 machines, and the tests work fine. Furthermore, Nils Wentzell reports that he was not able to reproduce the problems on his Intel Skylake.
Could it be specific to the configuration of that particular Skylake box? Can you rerun it to see if it is reproducible? Could it be a one-time quirk?


> On 10 Sep 2019, at 09:08, Kenneth Hoste <notifications@github.com> wrote:
> 
> @rokzitko <https://github.com/rokzitko> Latest test reports show that the tests work just fine on Intel Haswell and Broadwell, so the problem seems specific to Intel Skylake.
> Any idea what the cause could be?
> 
> If this is a known problem with TRIQS and there's an open issue to track it, I guess this doesn't need to block this PR...
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/easybuilders/easybuild-easyconfigs/pull/8835?email_source=notifications&email_token=AG57NBDXZMZIQB54A3CDL7TQI5BVJA5CNFSM4IPOUZWKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD6KCOLQ#issuecomment-529803054>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AG57NBB7YJKHDVOFRKO5P53QI5BVJANCNFSM4IPOUZWA>.
>

migueldiascosta commented:
@boegel the same two tests failed for me on skylake

rokzitko commented:
ok, this should be investigated then.

> On 10 Sep 2019, at 13:48, Miguel Dias Costa <notifications@github.com> wrote:
> 
> @boegel <https://github.com/boegel> the same two tests failed for me on skylake
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/easybuilders/easybuild-easyconfigs/pull/8835?email_source=notifications&email_token=AG57NBE2MVI6O2L7EDP4EB3QI6CRJA5CNFSM4IPOUZWKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD6KZXUQ#issuecomment-529898450>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AG57NBGW3RZTMADZMZ7VWILQI6CRJANCNFSM4IPOUZWA>.
>

boegel commented:
@Wentzell Tried that, is this the information you're looking for?

```
$ export CTEST_OUTPUT_ON_FAILURE=1 &&  make test
...

99% tests passed, 2 tests failed out of 238

Total Test time (real) = 2338.61 sec

The following tests FAILED:
    136 - g_om_nu_nup (SEGFAULT)
    237 - wrap_multivar (Failed)
Errors while running CTest
make: *** [test] Error 8
```

Output for two failing tests:

```
136/238 Test #136: g_om_nu_nup ......................***Exception: SegFault  0.32 sec
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from Gf
[ RUN      ] Gf.G_om_nu_nup
[node3110:108236] *** Process received signal ***
[node3110:108236] Signal: Segmentation fault (11)
[node3110:108236] Signal code:  (128)
[node3110:108236] Failing at address: (nil)
[node3110:108236] [ 0] /lib64/libpthread.so.0(+0xf5d0)[0x2b9fe5f115d0]
[node3110:108236] [ 1] /software/HDF5/1.10.2-foss-2018b/lib/libhdf5.so.101(H5S_select_hyperslab+0x2fc)[0x2b9fe73ca07c]
[node3110:108236] [ 2] /software/HDF5/1.10.2-foss-2018b/lib/libhdf5.so.101(H5Sselect_hyperslab+0xdb)[0x2b9fe73cbd6b]
[node3110:108236] [ 3] /tmp/easybuild_build/TRIQS/2.1.1/foss-2018b-Python-2.7.15/easybuild_obj/triqs/libtriqs.so(_ZN5triqs2h517dataspace_from_LSEibPKyS2_S2_S2_+0x13a)[0x2b9fe59d0fe0]
[node3110:108236] [ 4] /tmp/easybuild_build/TRIQS/2.1.1/foss-2018b-Python-2.7.15/easybuild_obj/triqs/libtriqs.so(_ZN5triqs6arrays7h5_impl15data_space_implENS1_17array_stride_infoEb+0xf37)[0x2b9fe59080b6]
[node3110:108236] [ 5] /tmp//easybuild_build/TRIQS/2.1.1/foss-2018b-Python-2.7.15/easybuild_obj/triqs/libtriqs.so(_ZN5triqs6arrays7h5_impl16write_array_implISt7complexIdEEEvNS_2h55groupERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEPKT_NS1_17array_stride_infoE+0x34)[0x2b9fe590dd6c]
[node3110:108236] [ 6] /tmp/easybuild_build/TRIQS/2.1.1/foss-2018b-Python-2.7.15/easybuild_obj/test/triqs/gfs/multivar/g_om_nu_nup(_ZN5triqs6arrays7h5_impl11write_arrayINS0_10array_viewISt7complexIdELi7EvLb0ELb1EEEEEvNS_2h55groupERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERKT_b+0x2e0)[0x42d9ab]
[node3110:108236] [ 7] /tmp/easybuild_build/TRIQS/2.1.1/foss-2018b-Python-2.7.15/easybuild_obj/test/triqs/gfs/multivar/g_om_nu_nup(_ZN5triqs6arrays8h5_writeINS0_5arrayISt7complexIdELi7EvEEEEN5boost11enable_if_cIXaasrNS0_26is_amv_value_or_view_classIT_EE5valuesrNS0_31has_scalar_or_string_value_typeIS9_vEE5valueEvE4typeENS_2h55groupERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERKS9_+0x1e6)[0x42dc60]
[node3110:108236] [ 8] /tmp/easybuild_build/TRIQS/2.1.1/foss-2018b-Python-2.7.15/easybuild_obj/test/triqs/gfs/multivar/g_om_nu_nup(_Z5rw_h5IN5triqs3gfs2gfINS1_17cartesian_productIJNS1_6imfreqES4_S4_EEENS1_13tensor_valuedILi4EEEEEET_RKS9_NSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESH_+0x6f4)[0x42e3d1]
[node3110:108236] [ 9] /tmp/easybuild_build/TRIQS/2.1.1/foss-2018b-Python-2.7.15/easybuild_obj/test/triqs/gfs/multivar/g_om_nu_nup(_ZN19Gf_G_om_nu_nup_Test8TestBodyEv+0x1557)[0x4186ad]
[node3110:108236] [10] /tmp/easybuild_build/TRIQS/2.1.1/foss-2018b-Python-2.7.15/easybuild_obj/google_test/libgtest.so(_ZN7testing8internal38HandleSehExceptionsInMethodIfSupportedINS_4TestEvEET0_PT_MS4_FS3_vEPKc+0x1a)[0x2b9fe56a4f69]
[node3110:108236] [11] /tmp/easybuild_build/TRIQS/2.1.1/foss-2018b-Python-2.7.15/easybuild_obj/google_test/libgtest.so(_ZN7testing8internal35HandleExceptionsInMethodIfSupportedINS_4TestEvEET0_PT_MS4_FS3_vEPKc+0x5b)[0x2b9fe56abf1f]
[node3110:108236] [12] /tmp/easybuild_build/TRIQS/2.1.1/foss-2018b-Python-2.7.15/easybuild_obj/google_test/libgtest.so(_ZN7testing4Test3RunEv+0xab)[0x2b9fe569ff1f]
[node3110:108236] [13] /tmp/easybuild_build/TRIQS/2.1.1/foss-2018b-Python-2.7.15/easybuild_obj/google_test/libgtest.so(_ZN7testing8TestInfo3RunEv+0xef)[0x2b9fe56a00ff]
[node3110:108236] [14] /tmp/easybuild_build/TRIQS/2.1.1/foss-2018b-Python-2.7.15/easybuild_obj/google_test/libgtest.so(_ZN7testing8TestCase3RunEv+0x9f)[0x2b9fe56a01a1]
[node3110:108236] [15] /tmp/easybuild_build/TRIQS/2.1.1/foss-2018b-Python-2.7.15/easybuild_obj/google_test/libgtest.so(_ZN7testing8internal12UnitTestImpl11RunAllTestsEv+0x175)[0x2b9fe56a2139]
[node3110:108236] [16] /tmp/easybuild_build/TRIQS/2.1.1/foss-2018b-Python-2.7.15/easybuild_obj/google_test/libgtest.so(_ZN7testing8internal38HandleSehExceptionsInMethodIfSupportedINS0_12UnitTestImplEbEET0_PT_MS4_FS3_vEPKc+0x1a)[0x2b9fe56a5bb6]
[node3110:108236] [17] /tmp/easybuild_build/TRIQS/2.1.1/foss-2018b-Python-2.7.15/easybuild_obj/google_test/libgtest.so(_ZN7testing8internal35HandleExceptionsInMethodIfSupportedINS0_12UnitTestImplEbEET0_PT_MS4_FS3_vEPKc+0x5b)[0x2b9fe56ac340]
[node3110:108236] [18] /tmp/easybuild_build/TRIQS/2.1.1/foss-2018b-Python-2.7.15/easybuild_obj/google_test/libgtest.so(_ZN7testing8UnitTest3RunEv+0x5b)[0x2b9fe569ff7d]
[node3110:108236] [19] /tmp/easybuild_build/TRIQS/2.1.1/foss-2018b-Python-2.7.15/easybuild_obj/test/triqs/gfs/multivar/g_om_nu_nup(main+0x53)[0x416bbb]
[node3110:108236] [20] /lib64/libc.so.6(__libc_start_main+0xf5)[0x2b9fe7deb495]
[node3110:108236] [21] /tmp/easybuild_build/TRIQS/2.1.1/foss-2018b-Python-2.7.15/easybuild_obj/test/triqs/gfs/multivar/g_om_nu_nup[0x416ab9]
[node3110:108236] *** End of error message ***
```

```
237/238 Test #237: wrap_multivar ....................***Failed    0.56 sec
Command for the test: python /tmp/easybuild_build/TRIQS/2.1.1/foss-2018b-Python-2.7.15/triqs-2.1.1/test/cpp2py/wrap_multivar.py
CMake Error at /tmp/easybuild_build/TRIQS/2.1.1/foss-2018b-Python-2.7.15/easybuild_obj/Config/run_test.cmake:26 (message):
  Error runing test 'wrap_multivar': ; command:
  python;/tmp/easybuild_build/TRIQS/2.1.1/foss-2018b-Python-2.7.15/triqs-2.1.1/test/cpp2py/wrap_multivar.py;
  shell output: Segmentation fault!
```

Wentzell commented:
@boegel Thank you for rerunning the build!
Given that both failures are related to a segfault we will have to look into this on our side in more detail. I will get back to you as soon as we know more!

Wentzell commented:
We were able to trace this problem back to hdf5 `1.10.2` within Foss2018b beeing compiled with `-march=native`. We further checked that with Foss2019a +  hdf5`1.10.5` this problem no longer persists and all tests pass. 

```python
easyblock = 'CMakeMake'

name = 'TRIQS'
version = '2.2-rc1'
versionsuffix = '-Python-%(pyver)s'

homepage = 'https://triqs.github.io/'
description = """
 TRIQS (Toolbox for Research on Interacting Quantum Systems) is a
 scientific project providing a set of C++ and Python libraries to
 develop new tools for the study of interacting quantum systems.
"""

docurls = ['https://triqs.github.io/triqs/%(version_major_minor)s.x/reference.html']
software_license = 'LicenseGPLv3'

toolchain = {'name': 'foss', 'version': '2019a'}
toolchainopts = {'pic': True, 'usempi': True}

sources = ['%(version)s.tar.gz']
source_urls = ['https://github.com/TRIQS/triqs/archive/']
checksums = ['cf4ce7ea7ba873a5bbb3fd53aa544106dc7e76970d5962ba19e27c240b5b1641']

dependencies = [
    ('Python', '2.7.15'),
    ('Boost', '1.70.0'),
    ('Clang', '8.0.0'),
    ('GMP', '6.1.2'),
    ('HDF5', '1.10.5'),
    ('Mako', '1.0.8'),
    ('h5py', '2.9.0')
]

builddependencies = [
    ('CMake', '3.13.3')
]

separate_build_dir = True

runtest = 'test'

sanity_check_paths = {
    'files': ['lib/libtriqs.%s' % SHLIB_EXT],
    'dirs': ['bin', 'include', 'lib', 'share'],
}

sanity_check_commands = ["python -c 'import pytriqs'"]

modextrapaths = {
    'PYTHONPATH': 'lib/python%(pyshortver)s/site-packages',
    'CMAKE_PREFIX_PATH': ['lib/cmake/triqs', 'lib/cmake/cpp2py']
}
modextravars = {'TRIQS_ROOT': '%(installdir)s'}

moduleclass = 'phys'
```

The reason we are no longer building the triqs `2.1.1` but rather the release candidate `2.2-rc1` is that Foss2019a imposes gcc 8 which is unfortunately not compatible with our `2.1.1` release. Unless we have some other means of enforcing a different HDF5 version within Foss2018b I suggest to keep this PR open until we have tagged the final `2.2.0` release for triqs (within the next two weeks) and to update the build script to use Foss2019a. What do you think?

Wentzell commented:
We have just tagged the `2.2.0` release of our library.

@rokzitko Could you update the pull request with the following foss2019a based eb script

```python
easyblock = 'CMakeMake'

name = 'TRIQS'
version = '2.2.0'
versionsuffix = '-Python-%(pyver)s'

homepage = 'https://triqs.github.io/'
description = """
 TRIQS (Toolbox for Research on Interacting Quantum Systems) is a
 scientific project providing a set of C++ and Python libraries to
 develop new tools for the study of interacting quantum systems.
"""

docurls = ['https://triqs.github.io/triqs/%(version_major_minor)s.x/reference.html']
software_license = 'LicenseGPLv3'

toolchain = {'name': 'foss', 'version': '2019a'}
toolchainopts = {'pic': True, 'usempi': True}

sources = ['%(version)s.tar.gz']
source_urls = ['https://github.com/TRIQS/triqs/archive/']
checksums = ['5f7fca3dd140d86e464ab18ff91fa16febad865344ec2c5b924434f507ef4038']

dependencies = [
    ('Python', '2.7.15'),
    ('Boost', '1.70.0'),
    ('Clang', '8.0.0'),
    ('GMP', '6.1.2'),
    ('HDF5', '1.10.5'),
    ('Mako', '1.0.8'),
    ('h5py', '2.9.0')
]

builddependencies = [
    ('CMake', '3.13.3')
]

separate_build_dir = True

runtest = 'test'

sanity_check_paths = {
    'files': ['lib/libtriqs.%s' % SHLIB_EXT],
    'dirs': ['bin', 'include', 'lib', 'share'],
}

sanity_check_commands = ["python -c 'import pytriqs'"]

modextrapaths = {
    'PYTHONPATH': 'lib/python%(pyshortver)s/site-packages',
    'CMAKE_PREFIX_PATH': ['lib/cmake/triqs', 'lib/cmake/cpp2py']
}
modextravars = {'TRIQS_ROOT': '%(installdir)s'}

moduleclass = 'phys'
```

rokzitko commented:
I have updated the easyconfig for TRIQS. What is the status with various TRIQS apps? Will there be 2.2.0 releases soon? I have easyconfigs for 2.1.1 releases of CTHYB, DFTTools and TPRF (as well as NFFT as dependency) that I can easily update as well.

Wentzell commented on easybuild/easyconfigs/t/TRIQS/TRIQS-2.2.0-foss-2019a-Python-2.7.15.eb:
```diff
@@ -0,0 +1,56 @@
+toolchainopts = {'pic': True, 'usempi': True}
+
+sources = ['%(version)s.tar.gz']
+source_urls = ['https://github.com/TRIQS/triqs/archive/']
```
Can you change this to
```suggestion
sources = ['triqs-%(version)s.tar.gz']
source_urls = ['https://github.com/TRIQS/triqs/releases/download/%(version)s/']
```

Wentzell replied:
This is meant to be a replacement of both lines, not just one. Somehow this wasn't picked up

Wentzell commented:
> I have updated the easyconfig for TRIQS. What is the status with various TRIQS apps? Will there be 2.2.0 releases soon? I have easyconfigs for 2.1.1 releases of CTHYB, DFTTools and TPRF (as well as NFFT as dependency) that I can easily update as well.

We hope to tag the applications as soon as possible!
Do you already have easybuild scripts for those that we could add additional pull requests for once the tags are down?

boegel commented:
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (1 easyconfigs in this PR)
node3102.skitty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/a82c60b9e937fc6f8ca297c41710c78a for a full test report.

boegel commented on easybuild/easyconfigs/t/TRIQS/TRIQS-2.2.0-foss-2019a-Python-2.7.15.eb:
```diff
@@ -0,0 +1,55 @@
+
+sanity_check_paths = {
+    'files': ['lib/libtriqs.%s' % SHLIB_EXT],
+    'dirs': ['bin', 'include', 'lib', 'share'],
```
@rokzitko I think it's worth enhancing this a little bit, to check for some specific binaries, and also check for `lib/python%(pyshortver)s/site-packages` rather than just `lib`:

```python
sanity_check_paths = {
    'files': ['bin/triqs++', 'lib/libtriqs.%s' % SHLIB_EXT],
    'dirs': ['include/triqs', 'lib/python%(pyshortver)s/site-packages', 'share'],
}
```

Maybe it's also worth adding `triqs++ --help` as an additional sanity check command?

```python
sanity_check_commands = [
    "triqs++ --help",
    "python -c 'import pytriqs'",
]

rokzitko commented:
> On 3 Oct 2019, at 17:03, Nils Wentzell <notifications@github.com> wrote:
> 
> I have updated the easyconfig for TRIQS. What is the status with various TRIQS apps? Will there be 2.2.0 releases soon? I have easyconfigs for 2.1.1 releases of CTHYB, DFTTools and TPRF (as well as NFFT as dependency) that I can easily update as well.
> 
> We hope to tag the applications as soon as possible!
> Do you already have easybuild scripts for those that we could add additional pull requests for once the tags are down?
> 

Yes, I have the easyconfigs for 2.1.1 and they work fine. I’ll wait for 2.2.0 tags, update the easyconfigs, and make the pull requests.

rokzitko commented:
I will add some more specific sanity checks, but more importantly, I now realized that there is a hidden run-time dependency, namely c++2py requires clang.cindex (typically installed via pip install clang). What is the best way to handle this dependency?

boegel commented:
@rokzitko The best way to handle that is to create an additional easyconfig file for the `clang` Python bindings (obviously with `Clang` as a dependency in there), using the `PythonPackage` easyblock (or `PythonBundle` if `clang` has additional dependencies).

Let us know if you need help with that. It's probably best to add that easyconfig through a separate PR...

rokzitko commented:
Yes, I think I’ll need help with this one. It turns out that “pip install clang” fetches python bindings that were bundled with clang 6.0, but were repackaged as a stand-alone package (here: https://pypi.org/project/clang/ <https://pypi.org/project/clang/>). It’s an unofficial package, which does seem to work with more recent versions of clang, too. 
Generally, one should use the bindings which are bundled with the clang sources, but are typically not installed during clang builds, it seems, and this holds true for Clang-8.0.0 easyconfig in EB. One solution would be to fix the clang easyconfig so that it installs the binding as well.


> On 4 Oct 2019, at 11:24, Kenneth Hoste <notifications@github.com> wrote:
> 
> @rokzitko <https://github.com/rokzitko> The best way to handle that is to create an additional easyconfig file for the clang Python bindings (obviously with Clang as a dependency in there), using the PythonPackage easyblock (or PythonBundle if clang has additional dependencies).
> 
> Let us know if you need help with that. It's probably best to add that easyconfig through a separate PR...
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/easybuilders/easybuild-easyconfigs/pull/8835?email_source=notifications&email_token=AG57NBF4EG6UIGOGCVZ5MBDQM4DWDA5CNFSM4IPOUZWKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEALCOHA#issuecomment-538322716>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AG57NBEFFPI63YCYWBOYAJ3QM4DWDANCNFSM4IPOUZWA>.
>

boegel commented:
@rokzitko We can certainly look into enabling the Python bindings in the `Clang` installation, but since this implies pulling in `Python` as a dependency, that should probably be a separate easyconfig file with appropriate `versionsuffix`.

Is there any documentation on how to build the `Clang` Python bindings?

Wentzell commented:
@boegel The libclang python bindings do not need any compilation.
Internally the module uses `ctypes` to call different functions in `libclang.so`.

When it comes to compatibility I am not quite sure which versions of llvm/clang the [clang pypi-package](https://pypi.org/project/clang/) is compatible against. I just compared the latest sources at https://github.com/llvm/llvm-project/tree/llvmorg-9.0.0/clang/bindings/python/clang against the ones contained in the pypi-package (6.0.0) and they seem to be largely identical.

Maybe you could just package the python bindings together with `Clang`  without adding `Python` as a dependency?

Wentzell commented:
I believe the installation of python bindings can be enabled during the llvm build by adding e.g.
`-DCLANG_PYTHON_BINDINGS_VERSIONS="2.7"` during the cmake configuration.

boegel commented:
@Wentzell The fact that a specific Python version needs to be specified implies that a `Python` dependency should be added, which I'm not a big fan of for `Clang`, for hopefully obvious reasons.

I wonder if there's an easy way to only install the Python bindings from a Clang source tarball (not using the unofficial PyPI package, it seems like that is likely to end in tears sooner or later)...

It seems like just copying the contents of `/clang/bindings/python` and updating `$PYTHONPATH` + setting `$CLANG_LIBRARY_PATH` could do it...

rokzitko commented:
Yes, I think this would be a sufficient (and possibly cleanest) solution. Which easyblock is the most appropriate for doing such things (copying specific files from tarballs)?

> On 5 Oct 2019, at 12:08, Kenneth Hoste <notifications@github.com> wrote:
> 
> @Wentzell <https://github.com/Wentzell> The fact that a specific Python version needs to be specified implies that a Python dependency should be added, which I'm not a big fan of for Clang, for hopefully obvious reasons.
> 
> I wonder if there's an easy way to only install the Python bindings from a Clang source tarball (not using the unofficial PyPI package, it seems like that is likely to end in tears sooner or later)...
> 
> It seems like just copying the contents of /clang/bindings/python and updating $PYTHONPATH + setting $CLANG_LIBRARY_PATH could do it...
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/easybuilders/easybuild-easyconfigs/pull/8835?email_source=notifications&email_token=AG57NBAHYLC4MEQIMF3RCBLQNBRTRA5CNFSM4IPOUZWKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEANPBGA#issuecomment-538636440>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AG57NBEOTIQ6KNTG3LKWGOLQNBRTRANCNFSM4IPOUZWA>.
>

boegel commented:
@rokzitko Probably `Tarball` easyblock, and you can probably use `start_dir = 'clang/bindings/python'` to specify that only the contents of that subdir should be copied.

For the environment variables, you can use `modextrapaths` for `$PYTHONPATH` (since that should be prepended) and `modextravars` for `$CLANG_LIBRARY_PATH` (figuring out the right value to set is going to be the tricky part I guess).

rokzitko commented:
The following seems to work (Clang-Python-bindings-8.0.0-GCCcore-8.2.0-Python-2.7.15.eb):

easyblock = 'Tarball'

name = 'Clang-Python-bindings'
version = '8.0.0'
versionsuffix = '-Python-%(pyver)s'

homepage = 'https://clang.llvm.org'
description = """Python bindings for libclang"""

toolchain = {'name': 'GCCcore', 'version': '8.2.0'}

source_urls = ["https://llvm.org/releases/%(version)s"]
sources = ['cfe-%(version)s.src.tar.xz']

dependencies = [
  ('Clang', version),
  ('Python', '2.7.15')
]

start_dir = 'bindings/python'

sanity_check_paths = {
  'files': ['clang/cindex.py'],
  'dirs': ['clang']
}

modextrapaths = {
  'PYTHONPATH': ''
}

moduleclass = 'lib'


If this looks acceptable, I'll make a separate PR. This is then used as a dependency in TRIQS:

dependencies = [
    ('Python', '2.7.15'),
    ('SciPy-bundle', '2019.03'),
    ('Boost', '1.70.0'),
    ('Clang', '8.0.0'),
    ('Clang-Python-bindings', '8.0.0', '-Python-2.7.15', ('GCCcore', '8.2.0')),
    ('GMP', '6.1.2'),
    ('HDF5', '1.10.5'),
    ('Mako', '1.0.8'),
    ('h5py', '2.9.0')
]

(I've also added SciPy-bundle, since TRIQS requires numpy.) 
I've also added a test for c++2py:

sanity_check_commands = [
    "triqs++ --help",
    "c++2py --help",
    "python -c 'import pytriqs'"
]

boegel commented:
@rokzitko No need to set `$CLANG_LIBRARY_PATH` to make the Clang Python bindings work as expected?

I think `clang` as name should suffice, the `-Python-*` versionsuffix will make it clear that it's the Python bindings.

You should also add an `import` test:

```python
sanity_check_commands = ["python -c 'import clang'"]
```

For the dependency specification, there's no need to hardcode the toolchain, this will work just fine:

```
```

rokzitko commented:
> On 5 Oct 2019, at 18:22, Kenneth Hoste <notifications@github.com> wrote:
> 
> @rokzitko <https://github.com/rokzitko> No need to set $CLANG_LIBRARY_PATH to make the Clang Python bindings work as expected?
> 
I think the path to libclang.so in LD_LIBRARY_PATH (set in Clang module) is sufficient.
> I think clang as name should suffice, the -Python-* versionsuffix will make it clear that it's the Python bindings.
> 
But wouldn't this suggest that this is clang compiler with Python bindings, rather than solely Python bindings for clang from another module?
Also, if I rename it to clang alone, I get the following error:
ERROR:150: Module 'Clang/8.0.0-GCCcore-8.2.0-Python-2.7.15' conflicts with the currently loaded module(s) 'Clang/8.0.0-GCCcore-8.2.0'
I thus propose to keep the clang-Python-bindings name, and reserve the name clang-x.x.x-Python-* for possible future easyconfig for clang with bundled bindings.
> You should also add an import test:
> 
> sanity_check_commands = ["python -c 'import clang'"]
Ok.
> For the dependency specification, there's no need to hardcode the toolchain, this will work just fine:
> 
> 
Ok.

boegel commented:
@rokzitko I overlooked the name clash issue, so OK, let's go with `Clang-Python-bindings`.

boegel commented:
@rokzitko #9084 is merged, so this still needs to be updated to include `Clang-Python-bindings` as a dependency?

rokzitko commented:
Yes, I have just pushed the update.

boegel commented:
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node3103.skitty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/447eb9d479f66197783045d22fe4807e for a full test report.

Wentzell commented:
Great. Thanks a lot @rokzitko for making this work!

boegel commented:
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2696.swalot.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/ac8acfc9a50877901c44cf88fd9822d0 for a full test report.

boegel commented:
Going in, thanks @rokzitko!

boegel commented:
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
node2106.delcatty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/711c8c298fb68759af5ca7084874670e for a full test report.
